pkg/apis/migration/migration.go

```diff
@@ -0,0 +1,351 @@
+			if err != nil {
+				log.Errorf("failed to create dynamic informer for volume migration CRD. Err: %v", err)
+				return
+			}
+			handlers := cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {
+					log.Debugf("received add event for VolumeMigration CR!")
+					var volumeMigrationObject migrationv1alpha1.CnsVSphereVolumeMigration
+					runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object, &volumeMigrationObject)
+					volumeMigrationInstance.volumePathToVolumeID.Store(volumeMigrationObject.Spec.VolumePath, volumeMigrationObject.Spec.VolumeID)
+					log.Debugf("successfully added volumePath: %q, volumeID: %q mapping in the cache", volumeMigrationObject.Spec.VolumePath, volumeMigrationObject.Spec.VolumeID)
+				},
+				DeleteFunc: func(obj interface{}) {
+					log.Debugf("received delete event for VolumeMigration CR!")
+					var volumeMigrationObject migrationv1alpha1.CnsVSphereVolumeMigration
+					runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object, &volumeMigrationObject)
+					volumeMigrationInstance.volumePathToVolumeID.Delete(volumeMigrationObject.Spec.VolumePath)
+					log.Debugf("successfully deleted volumePath: %q, volumeID: %q mapping from cache", volumeMigrationObject.Spec.VolumePath, volumeMigrationObject.Spec.VolumeID)
+				},
+			}
+			informer.Informer().AddEventHandler(handlers)
+			stopCh := make(chan struct{})
+			informer.Informer().Run(stopCh)
+		}()
+	})
+	if volumeMigrationServiceInitErr != nil {
+		log.Info("volume migration service initialization failed")
+		return nil, volumeMigrationServiceInitErr
+	}
+	log.Info("volume migration service initialized")
+	return volumeMigrationInstance, nil
+}
+
+// GetVolumeID returns VolumeID for given VolumePath
+// Returns an error if not able to retrieve VolumeID.
+func (volumeMigration *volumeMigration) GetVolumeID(ctx context.Context, volumePath string) (string, error) {
+	log := logger.GetLogger(ctx)
+	info, found := volumeMigration.volumePathToVolumeID.Load(volumePath)
+	if found {
+		log.Infof("VolumeID: %q found from the cache for VolumePath: %q", info.(string), volumePath)
+		return info.(string), nil
+	}
+	log.Infof("Could not retrieve Volume ID from cache for Volume Path: %q. volume may not be registered. Registering Volume with CNS", volumePath)
+	volumeID, err := volumeMigration.registerVolume(ctx, volumePath)
+	if err != nil {
+		log.Error("failed to register volume volume: %q, err: %v", volumePath, err)
+		return "", err
+	}
+	log.Infof("successfully registered volume: %q with CNS. VolumeID: %q", volumePath, volumeID)
+	cnsvSphereVolumeMigration := migrationv1alpha1.CnsVSphereVolumeMigration{
+		ObjectMeta: metav1.ObjectMeta{Name: volumeID},
+		Spec: migrationv1alpha1.CnsVSphereVolumeMigrationSpec{
+			VolumePath: volumePath,
+			VolumeID:   volumeID,
+		},
+	}
+	log.Infof("Saving cnsvSphereVolumeMigration CR: %v", cnsvSphereVolumeMigration)
```

Comment:
This can be log.Debugf()
And in line: 184 we can have:
<pre>
 log.Infof("successfully saved cnsvSphereVolumeMigration CR: %v", cnsvSphereVolumeMigration)
</pre>

Reply:
yes marked as Debug and added Info level log after operation is sucessful.